code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,312 @@
+	spawning = 3
+	lifespan = 9
+	fade = 12
+	grow = 0.04
+	velocity = list(0, 0)
+	position = generator("circle", 16, 16, NORMAL_RAND)
+	drift = generator("vector", list(0, -0.15), list(0, 0.15))
+	gravity = list(0, 0.8)
+	scale = generator("vector", list(0.1, 0.1), list(0.6,0.6), NORMAL_RAND)
+	rotation = 0
+	spin = generator("num", 10, 20)
+
+/datum/status_effect/drone_enhancement
+	id = "drone enhancement"
+	duration = -1
+	tick_interval = 2 SECONDS
+	alert_type = null
+	/// Used to track the giver of this buff.
+	var/mob/living/carbon/xenomorph/buffing_target
+	/// Used to track the owner of this buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+	/// Used to determine whether there is an existing Essence Link or not. Also allows access to its vars.
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// The plasma cost per tick of this ability.
+	var/plasma_cost
+	/// Damage bonus given by this ability.
+	var/damage_addition
+	/// Speed bonus given by this ability.
+	var/speed_addition
+
+/datum/status_effect/drone_enhancement/on_creation(mob/living/new_owner, mob/living/carbon/new_target)
+	buff_owner = new_owner
+	buffing_target = new_target
+	existing_link = buffing_target.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	return ..()
+
+/datum/status_effect/drone_enhancement/on_apply()
+	plasma_cost = round(buffing_target.xeno_caste.plasma_max * 0.1) // 10% of the buff giver's maximum plasma
+	damage_addition = round(buff_owner.xeno_caste.melee_damage * 0.04) // 4% of the user's base damage APPLIED AS A MULTIPLIER.
+	speed_addition = round(buff_owner.xeno_caste.speed * 0.05) // 5% of the user's base speed.
+	buffing_target.xeno_caste.plasma_gain = 0 // Denies natural plasma gain while this is active.
+
+	toggle_particles(TRUE)
+	buff_owner.xeno_melee_damage_modifier = damage_addition
+	buff_owner.add_movespeed_modifier(MOVESPEED_ID_ENHANCEMENT, TRUE, 0, NONE, FALSE, speed_addition)
+	buff_owner.balloon_alert(buff_owner, "Enhancement active")
+	buffing_target.balloon_alert(buffing_target, "Enhancement active")
+	buff_owner.visible_message(span_xenowarning("[buff_owner] is engulfed in a burst of plasma!"))
+	INVOKE_ASYNC(buff_owner, /mob/living/carbon/xenomorph/.proc/emote, "roar2")
+	RegisterSignal(buff_owner, COMSIG_MOB_DEATH, .proc/handle_death)
+	RegisterSignal(buffing_target, COMSIG_MOB_DEATH, .proc/handle_death)
+	return TRUE
+
+/datum/status_effect/drone_enhancement/on_remove()
+	buff_owner.balloon_alert(buff_owner, "Enhancement inactive")
+	buffing_target.balloon_alert(buffing_target, "Enhancement inactive")
+	buff_owner.xeno_melee_damage_modifier = initial(buff_owner.xeno_melee_damage_modifier)
+	buff_owner.remove_movespeed_modifier(MOVESPEED_ID_ENHANCEMENT)
+	buffing_target.xeno_caste.plasma_gain = initial(buffing_target.xeno_caste.plasma_gain)
```

Comment:
You might want to use -= instead of initial here in case something else modifies these variables

Reply:
Done in next commit.